submissions/NJul/js-oop/app.js

```diff
@@ -0,0 +1,143 @@
+const Inhabitant = function (x, y, speed) {
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+};
+
+// Renders the enemy and the player into the game
+Inhabitant.prototype.render = function () {
+  ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Inhabitant.prototype.update = function (dt) {};
+
+// Enemies our player must avoid
+const Enemy = function (x, y, speed) {
+  Inhabitant.call(this, x, y, speed);
+  // The following variables are used to determine the x and y axis and speed of the enemy
+  this.x = x;
+  this.y = y;
+  this.speed = speed;
+
+  // The image of the enemy of cockroach that is added to the playing field
+  this.sprite = 'images/enemy-bug.png';
+};
+
+Enemy.prototype = Object.create(Inhabitant.prototype);
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+Enemy.prototype.update = function (dt) {
+  // Multiplies the speed by the dt parameter on the x axis
+  this.x += this.speed * dt;
+
+  // Once enemies are off the canvas, they reappear randomly with different speeds
+  if (this.x > 510) {
+    this.x = -50;
+    this.speed = 100 + Math.floor(Math.random() * 400);
+  }
+
+  // Checks for collisions between the player and the enemies
+  if (
+    player.x < this.x + 80 &&
+    player.x + 80 > this.x &&
+    player.y < this.y + 60 &&
+    60 + player.y > this.y
+  ) {
+    player.x = 202;
+    player.y = 405;
+    score.updateMisses();
+  }
+};
+
+// Player
+const Player = function (x, y) {
+  Inhabitant.call(this, x, y);
+  // The following variables are used to determine the x and y axis
+  this.x = x;
+  this.y = y;
+
+  // The image of the player that is added to the playing field
+  this.sprite = 'images/char-horn-girl.png';
+};
+
+Player.prototype = Object.create(Inhabitant.prototype);
+
+// Allows the user to use the arrow keys to jump from tile to tile
+Player.prototype.handleInput = function (keyPress) {
+  // Enables user on left arrow key to move left on the x axis by 102
+  // Also enables user not to go off the game tiles on the left side
+  if (keyPress == 'left' && this.x > 0) {
+    this.x -= 102;
+  }
+
+  // Enables user on right arrow key to move right on the x axis by 102
+  // Also enables user not to go off the game tiles on the right side
+  if (keyPress == 'right' && this.x < 405) {
+    this.x += 102;
+  }
+
+  // Enables user on up arrow key to move upwards on the y axis by 83
+  if (keyPress == 'up' && this.y > 0) {
+    this.y -= 83;
+  }
+
+  // Enables user on down arrow key to move downwards on the y axis by 83
+  // Also enables user not to go off the game tiles on the bottom side
+  if (keyPress == 'down' && this.y < 405) {
+    this.y += 83;
+  }
+
+  // Once the user reaches the top of the page; the water, the user is
+  // Instantly reset to the starting position
+  if (this.y < 1) {
+    this.x = 202;
```

Comment:
`202`, `405` and all other numbers in the code look mysterious.

All numbers like block dimensions, initial locations should be defined as named constants (e.g. `const STEP = 101;`) as otherwise numbers scattered across code base look cryptic.

Named constants add semantic meaning and improve readability. Every number that has a semantic purpose (like those listed above) should be defined as a constant.

Think of how your code **reads** - the closer to plain English the better.

Reply:
Thank you for your review. I decided to learn now at my own pace. Happy New Year!